src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -380,7 +380,9 @@ private void iterablesRemove(Iterable<?> actual, Object value) {
    */
   public void assertContainsOnly(AssertionInfo info, Iterable<?> actual, Object[] expectedValues) {
     final List<?> actualAsList = newArrayList(actual);
-    if (commonCheckThatIterableAssertionSucceeds(info, actualAsList, expectedValues)) return;
+    checkNotNullIterables(info, actualAsList, expectedValues);
+    // if both actual and values are empty, then assertion passes.
+    if (actualAsList.isEmpty() && expectedValues.length == 0) return;
```

Comment:
@jhonnen how does that address the issue?
`commonCheckThatIterableAssertionSucceeds` was doing the same thing with:
```
if (!actual.iterator().hasNext() && sequence.length == 0) return true;
```
You would need to add a test showing the improved behavior.

Reply:
The issue was not caused by this check, but by `failIfEmptySinceActualIsNotEmpty(sequence)` from `commonCheckThatIterableAssertionSucceeds`.